client/my-sites/site-settings/seo-settings/main.jsx

```diff
@@ -1,61 +1,28 @@
-import PropTypes from 'prop-types';
-import { Component } from 'react';
-import { connect } from 'react-redux';
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import AsyncLoad from 'calypso/components/async-load';
 import QuerySitePurchases from 'calypso/components/data/query-site-purchases';
-import QuerySiteSettings from 'calypso/components/data/query-site-settings';
 import { errorNotice } from 'calypso/state/notices/actions';
-import {
-	getSitePurchases,
-	hasLoadedSitePurchasesFromServer,
-	getPurchasesError,
-} from 'calypso/state/purchases/selectors';
+import { getPurchasesError } from 'calypso/state/purchases/selectors';
 import { getSelectedSiteId } from 'calypso/state/ui/selectors';
 
-export class SeoSettings extends Component {
-	// @TODO: Please update https://github.com/Automattic/wp-calypso/issues/58453 if you are refactoring away from UNSAFE_* lifecycle methods!
-	UNSAFE_componentWillReceiveProps( nextProps ) {
-		if ( ! this.props.purchasesError && nextProps.purchasesError ) {
-			this.props.errorNotice( nextProps.purchasesError );
-		}
-	}
+export function SeoSettings() {
+	const dispatch = useDispatch();
+	const siteId = useSelector( getSelectedSiteId );
+	const purchasesError = useSelector( getPurchasesError );
 
-	render() {
-		const { siteId } = this.props;
+	useEffect( () => {
+		if ( purchasesError ) {
+			dispatch( errorNotice( purchasesError ) );
+		}
+	}, [ dispatch, purchasesError ] );
 
-		return (
-			<div>
-				<QuerySiteSettings siteId={ siteId } />
```

Comment:
We don't make any use of site settings in this component. The `SeoSettingsForm` component (the only relevant child) is rendering `<QuerySiteSettings />` on its own.